File: swift-bridge/Sources/SmixSDK/SystemPopup.swift
// SystemPopup + SystemPopupButton — mirror the smix-runner-wire
// `/system-popups` shape returned by `SmixSession.systemPopups()`
// (`#[serde(rename_all = "camelCase")]`).

import Foundation

/// One system popup on screen (alert / sheet / banner) — the shape the
/// FFI `SmixSession.systemPopups()` returns.
public struct SystemPopup: Codable, Sendable, Equatable {
    /// Stable identifier for matching across polls.
    public let id: String
    /// Discriminator (e.g. `"alert"` / `"sheet"` / `"banner"`).
    public let type: String
    /// Originator (e.g. bundle id, system framework name).
    public let source: String
    /// Popup title text.
    public let title: String
    /// Popup body text.
    public let body: String
    /// Buttons available on the popup.
    public let buttons: [SystemPopupButton]

    public init(
        id: String,
        type: String,
        source: String,
        title: String = "",
        body: String = "",
        buttons: [SystemPopupButton] = []
    ) {
        self.id = id
        self.type = type
        self.source = source
        self.title = title
        self.body = body
        self.buttons = buttons
    }
}

/// One button on a [`SystemPopup`].
public struct SystemPopupButton: Codable, Sendable, Equatable {
    /// Stable identifier for the button.
    public let id: String
    /// Visible button label.
    public let label: String
    /// Semantic role (`"cancel"` / `"destructive"` / `"default"`).
    public let role: String
    /// Whether tapping this button performs a destructive action.
    public let dangerous: Bool
    /// Optional outcome hint (e.g. `"grants location permission"`).
    public let outcomeHint: String?

    public init(
        id: String,
        label: String,
        role: String,
        dangerous: Bool = false,
        outcomeHint: String? = nil
    ) {
        self.id = id
        self.label = label
        self.role = role
        self.dangerous = dangerous
        self.outcomeHint = outcomeHint
    }
}

File: android-runner/sdk/src/main/kotlin/dev/smix/sdk/SystemPopup.kt
// SystemPopup + SystemPopupButton — mirror the smix-runner-wire
// SystemPopup shape returned as a JSON array by
// `SmixSession.systemPopups()` (`#[serde(rename_all = "camelCase")]`,
// `type` renamed).

package dev.smix.sdk

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * One system popup on screen (alert / sheet / banner) — the shape the
 * FFI `SmixSession.systemPopups()` returns.
 */
@Serializable
data class SystemPopup(
    @SerialName("id") val id: String,
    @SerialName("type") val type: String,
    @SerialName("source") val source: String,
    @SerialName("title") val title: String = "",
    @SerialName("body") val body: String = "",
    @SerialName("buttons") val buttons: List<SystemPopupButton> = emptyList(),
)

/** One button on a [SystemPopup]. */
@Serializable
data class SystemPopupButton(
    @SerialName("id") val id: String,
    @SerialName("label") val label: String,
    @SerialName("role") val role: String,
    @SerialName("dangerous") val dangerous: Boolean = false,
    @SerialName("outcomeHint") val outcomeHint: String? = null,
)
